### PR TITLE
#1654 added regex to is_inconsistent_across_dataset operation

### PR DESCRIPTION
**Pull request by alexfurmenkov**

### CODE REVIEW

File: tests/unit/test_check_operators/test_value_set_checks.py [hunk 1]
```diff
@@ -244,6 +244,47 @@ def test_is_inconsistent_across_dataset(
     assert result.equals(df.convert_to_series(expected_result))
 
 
+def test_is_inconsistent_across_dataset_regex():
```
Comment (RamilCDISC) [inline tests/unit/test_check_operators/test_value_set_checks.py]: could you please cover multiple cases such as regex missing, edge cases around regex shape etc.

File: cdisc_rules_engine/check_operators/dataframe_operators.py [hunk 2]
```diff
@@ -1162,6 +1165,17 @@ def is_inconsistent_across_dataset(self, other_value):
                     grouping_cols.append(col)
         df_check = self.value[grouping_cols + [target]].copy()
         df_check = df_check.fillna("_NaN_")
+        if regex:
+            pattern = re.compile(regex)
+            if pattern.groups == 0:
+                regex = f"({regex})"
+            extracted = df_check[target].astype(str).str.extract(regex, expand=False)
```
Comment (RamilCDISC) [inline cdisc_rules_engine/check_operators/dataframe_operators.py]: The documentation mentions that only first capture group will be used but here whole regex is applied, which will return multiple columns if there are more than 1 groups. Could you please reconcile these? Also could you then cover the case of multiple groups and invalid regex in tests.

**Comment by alexfurmenkov** [comment]
@RamilCDISC hi, regarding last message, can you please re-run your test but with the fixed rule? I've made a mistake providing you examle rule. according to documentation,operation should look like that:

```YAML
- name: "BGSTRESU"
  operator: is_inconsistent_across_dataset
  value: "USUBJID"
```

 or with multiple grouping values
```YAML
- name: "--STRESU"
  operator: is_inconsistent_across_dataset
  value:
    - "--TESTCD"
    - "--CAT"
    - "--SCAT"
    - "--SPEC"
    - "--METHOD" 
```
here is updated rule 
[Rule.yml](https://github.com/user-attachments/files/26121853/Rule.yml)
and the report after running the rule
[CORE-Report-2026-03-19T17-34-49.json](https://github.com/user-attachments/files/26121893/CORE-Report-2026-03-19T17-34-49.json)

File: cdisc_rules_engine/check_operators/dataframe_operators.py [hunk 3]
```diff
@@ -1162,6 +1166,21 @@ def is_inconsistent_across_dataset(self, other_value):
                     grouping_cols.append(col)
         df_check = self.value[grouping_cols + [target]].copy()
         df_check = df_check.fillna("_NaN_")
+        if regex:
+            try:
+                pattern = re.compile(regex)
+            except re.error:
+                logger.error("Invalid regex: %s. It will be ignored.", regex)
+                regex, pattern = r".*", re.compile(r".*")
```
Comment (gerrycampion) [inline cdisc_rules_engine/check_operators/dataframe_operators.py]: I think an invalid regex should trickle up as an execution error so that the rule author has to fix it.
